File: domain/src/main/scala/com/anjunar/technologyspeaks/codemirror/CodeMirrorCSS.scala
package com.anjunar.technologyspeaks.codemirror

import com.anjunar.scala.mapper.annotations.PropertyDescriptor
import jakarta.persistence.{Basic, Column, Entity, Lob}

import scala.compiletime.uninitialized

@Entity
class CodeMirrorCSS extends AbstractCodeMirrorFile {

  @Basic
  @Lob
  @Column(columnDefinition = "text")
  @PropertyDescriptor(title = "Content", writeable = true)
  var content: String = uninitialized

}

File: application/src/main/typescript/packages/library/src/components/inputs/codemirror/domain/CodeMirrorCSS.ts
import {AbstractCodeMirrorFile} from "./AbstractCodeMirrorFile";
import Basic from "../../../../mapper/annotations/Basic";
import {Entity} from "../../../../mapper";
import {CodeMirrorContent} from "./CodeMirrorContent";

@Entity("CodeMirrorCSS")
export class CodeMirrorCSS extends AbstractCodeMirrorFile implements CodeMirrorContent {

    $type = "CodeMirrorCSS"

    @Basic()
    content : string

}